strawberryfields/engine.py

```diff
@@ -531,161 +544,542 @@ def run(self, program, *, args=None, compile_options=None, run_options=None):
+            job_id (str): the job ID
 
         Returns:
-            (list): a list representing the result samples
+            strawberryfields.engine.Result: the job result
+        """
+        response = self._get(
+            "/jobs/{}/result".format(job_id), {"Accept": "application/x-numpy"},
+        )
+        if response.status_code == 200:
+            # Read the numpy binary data in the payload into memory
+            with io.BytesIO() as buf:
+                buf.write(response.content)
+                buf.seek(0)
+                samples = np.load(buf)
+            return Result(samples, is_stateful=False)
+        raise RequestFailedError(self._format_error_message(response))
+
+    def cancel_job(self, job_id: str):
+        """Cancels a job.
 
-        Raises:
-            Exception: In case a job could not be submitted or completed.
-            TypeError: In case a job is already queued and a user is trying to submit a new job.
+        Args:
+            job_id (str): the job ID
         """
-        if self.jobs:
-            raise OneJobAtATimeError(
-                "A job is already queued. Please reset the engine and try again."
-            )
+        response = self._patch(
+            "/jobs/{}".format(job_id), data={"status", JobStatus.CANCELLED.value}
+        )
+        if response.status_code == 204:
+            return
+        raise RequestFailedError(self._format_error_message(response))
 
-        kwargs.update(program.run_options)
-        job_content = self._get_blackbird(program=program, **kwargs).serialize()
-        job = self._queue_job(job_content)
+    def ping(self) -> bool:
+        """Tests the connection to the remote backend.
 
-        try:
-            while not job.is_failed and not job.is_complete:
-                job.reload()
-                time.sleep(self.polling_delay_seconds)
-        except KeyboardInterrupt:
-            if job.id:
-                print("Job {} is queued in the background.".format(job.id.value))
-            else:
-                self.reset()
-                raise JobNotQueuedError("Job was not sent to server. Please try again.")
+        Returns:
+            bool: ``True`` if the connection is successful, and ``False`` otherwise
+        """
+        response = self._get("/healthz")
+        return response.status_code == 200
+
+    def _get(
+        self, path: str, headers: Dict[str, str] = None, **kwargs
+    ) -> requests.Response:
+        return self._request(RequestMethod.GET, path, headers, **kwargs)
+
+    def _post(
+        self, path: str, headers: Dict[str, str] = None, **kwargs
+    ) -> requests.Response:
+        return self._request(RequestMethod.POST, path, headers, **kwargs)
+
+    def _patch(
+        self, path: str, headers: Dict[str, str] = None, **kwargs
+    ) -> requests.Response:
+        return self._request(RequestMethod.PATCH, path, headers, **kwargs)
```

Comment:
Maybe these could be made static, what do you think?

Reply:
Went one better and removed them completely as we now use the `requests` functions directly :slightly_smiling_face: